### ISSUE TITLE
the result is different every time

### ISSUE THREAD

**Issue by weickdev** [dgraph-io/dgraph#1783]
I change the query on the http://docs.dgraph.io, when I run the below query
```
{
  brCharacters(func: eq(name@en, "Blade Runner")) {
    name@en
    initial_release_date
    starring {
      expand(_all_){
        name@en
      }
    }
  }
}
```

First time I run this query, I get below result:
```
{
  "data": {
    "brCharacters": [
      {
        "name@en": "Blade Runner",
        "initial_release_date": "1982-06-25T00:00:00Z",
        "starring": [
          {
            "performance.film": [
              {
                "name@en": "Blade Runner"
              }
            ]
          },
          {
            "performance.film": [
              {
                "name@en": "Blade Runner"
              }
            ]
          },
         ....
    ]
  }
}
```
All the elements of starring are performance.film, then I run it again, I get below:
```
{
  "data": {
    "brCharacters": [
      {
        "name@en": "Blade Runner",
        "initial_release_date": "1982-06-25T00:00:00Z",
        "starring": [
          {
            "performance.character": [
              {
                "name@en": "Bear"
              }
            ]
          },
          {
            "performance.character": [
              {
                "name@en": "Pris"
              }
            ]
          },
         ....         
        ]
      }
    ]
  }
}
```
All the elements of starring are performance.character. It seems it only expands one edge each time. Shouldn't it expands all the edges?

**Comment by pawanrawal** [comment]
Thanks for filling the issue @weickdev. I have investigated the issue and it looks like a race condition. I will have a fix for this soon.

**Comment by pawanrawal** [comment]
This is fixed with fe424a1. Next release would contain the fix.
